README.md
# REFormattedNumberField

UITextField subclass that allows numeric input in a predefined format.

![Screenshot of REFormattedNumberField](https://github.com/romaonthego/REFormattedNumberField/raw/master/Screenshot.png "REFormattedNumberField Screenshot")

## Requirements
* Xcode 4.5 or higher
* Apple LLVM compiler
* iOS 5.0 or higher
* ARC

## Demo

Build and run the `REFormattedNumberFieldExample` project in Xcode to see `REFormattedNumberField` in action.

## Installation

### CocoaPods

The recommended approach for installating `REFormattedNumberField` is via the [CocoaPods](http://cocoapods.org/) package manager, as it provides flexible dependency management and dead simple installation.
For best results, it is recommended that you install via CocoaPods >= **0.15.2** using Git >= **1.8.0** installed via Homebrew.

Install CocoaPods if not already available:

``` bash
$ [sudo] gem install cocoapods
$ pod setup
```

Change to the directory of your Xcode project:

``` bash
$ cd /path/to/MyProject
$ touch Podfile
$ edit Podfile
```

Edit your Podfile and add REFormattedNumberField:

``` bash
platform :ios, '5.0'
pod 'REFormattedNumberField', '~> 1.0.2'
```

Install into your Xcode project:

``` bash
$ pod install
```

Open your project in Xcode from the .xcworkspace file (not the usual project file)

``` bash
$ open MyProject.xcworkspace
```

Please note that if your installation fails, it may be because you are installing with a version of Git lower than CocoaPods is expecting. Please ensure that you are running Git >= **1.8.0** by executing `git --version`. You can get a full picture of the installation details by executing `pod install --verbose`.

### Manual Install

All you need to do is drop `REFormattedNumberField` files into your project, and add `#include "REFormattedNumberField.h"` to the top of classes that will use it.

## Example Usage

Pretty much all you need is just to specify `format` NSString, where `X` are digits and all other characters are separators.

``` objective-c
REFormattedNumberField *phoneField1 = [[REFormattedNumberField alloc] initWithFrame:CGRectMake(20, 20, 280, 30)];
phoneField1.format = @"(XXX) XXX-XXXX";
[self.view addSubview:phoneField1];

REFormattedNumberField *phoneField2 = [[REFormattedNumberField alloc] initWithFrame:CGRectMake(20, 80, 280, 30)];
phoneField2.format = @"+X (XXX) XXX-XXXX";
[self.view addSubview:phoneField2];

REFormattedNumberField *ccField = [[REFormattedNumberField alloc] initWithFrame:CGRectMake(20, 140, 280, 30)];
ccField.format = @"XXXX XXXX XXXX XXXX";
[self.view addSubview:ccField];

REFormattedNumberField *ccExpirationField = [[REFormattedNumberField alloc] initWithFrame:CGRectMake(20, 200, 280, 30)];
ccExpirationField.format = @"XX/XX";
[self.view addSubview:ccExpirationField];
```

## Contact

Roman Efimov

- https://github.com/romaonthego
- https://twitter.com/romaonthego
- romefimov@gmail.com

## License

REFormattedNumberField is available under the MIT license.

Copyright © 2013 Roman Efimov.

Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.


CHANGELOG.md
## 1.0.4
[REFormattedNumberField](https://github.com/romaonthego/REFormattedNumberField/compare/1.0.3...1.0.4)

###### Enhancements

* Add awakeFromNib support for Interface Builder. [#3](https://github.com/romaonthego/REFormattedNumberField/issues/3)

## 1.0.3
[REFormattedNumberField](https://github.com/romaonthego/REFormattedNumberField/compare/1.0.2...1.0.3)

###### Enhancements

* Expose `- (NSString *)string:(NSString *)string withNumberFormat:(NSString *)format;` in headers.

## 1.0.2
[REFormattedNumberField](https://github.com/romaonthego/REFormattedNumberField/compare/1.0.1...1.0.2)

###### Enhancements

* Add accessor for unformatted string. [#2](https://github.com/romaonthego/REFormattedNumberField/issues/2)

## 1.0.1
[REFormattedNumberField](https://github.com/romaonthego/REFormattedNumberField/compare/1.0...1.0.1)

###### Bug fixes

* Fix main thread blocking when updating the textfield. [#1](https://github.com/romaonthego/REFormattedNumberField/issues/1)

## 1.0

Initial release.

REFormattedNumberFieldExample/REFormattedNumberFieldExample/RootViewController.m
//
//  RootViewController.m
//  REFormattedNumberFieldExample
//
//  Created by Roman Efimov on 2/21/13.
//  Copyright (c) 2013 Roman Efimov. All rights reserved.
//

#import "RootViewController.h"
#import "REFormattedNumberField.h"

@implementation RootViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.view.backgroundColor = [UIColor colorWithWhite:0.9 alpha:1];
	
    REFormattedNumberField *phoneField1 = [[REFormattedNumberField alloc] initWithFrame:CGRectMake(20, 20, 280, 30)];
    phoneField1.backgroundColor = [UIColor whiteColor];
    phoneField1.format = @"(XXX) XXX-XXXX";
    phoneField1.placeholder = @"(123) 456-7890";
    phoneField1.font = [UIFont systemFontOfSize:21];
    [self.view addSubview:phoneField1];
    
    REFormattedNumberField *phoneField2 = [[REFormattedNumberField alloc] initWithFrame:CGRectMake(20, 80, 280, 30)];
    phoneField2.backgroundColor = [UIColor whiteColor];
    phoneField2.format = @"+X (XXX) XXX-XXXX";
    phoneField2.placeholder = @"+1 (123) 456-7890";
    phoneField2.font = [UIFont systemFontOfSize:21];
    [self.view addSubview:phoneField2];
    
    REFormattedNumberField *ccField = [[REFormattedNumberField alloc] initWithFrame:CGRectMake(20, 140, 280, 30)];
    ccField.backgroundColor = [UIColor whiteColor];
    ccField.format = @"XXXX XXXX XXXX XXXX";
    ccField.placeholder = @"1234 5678 9102 3456";
    ccField.font = [UIFont systemFontOfSize:21];
    [self.view addSubview:ccField];
    
    REFormattedNumberField *ccExpirationField = [[REFormattedNumberField alloc] initWithFrame:CGRectMake(20, 200, 280, 30)];
    ccExpirationField.backgroundColor = [UIColor whiteColor];
    ccExpirationField.format = @"XX/XX";
    ccExpirationField.placeholder = @"MM/YY";
    ccExpirationField.font = [UIFont systemFontOfSize:21];
    [self.view addSubview:ccExpirationField];
}


@end
